### PR TITLE
Replace Less with CSS

### LINKED ISSUE #15565: Replace Less
Less is a rather dead project which does not support recent CSS features as seen in https://github.com/go-gitea/gitea/pull/15520. I suggest moving to either SCSS or plain CSS.

Thoughts on SCSS:

 - Supports pretty much everything and more that Less does
 - Is 4 times more populare then Less
 - Might be slower to compile with their [Dart implementation](https://github.com/sass/dart-sass)
 - Like less, it encourages the bad practice of deeply nesting selectors, resulting in unnecessarily big CSS output

Thoughts on CSS:

- Instant build, no compilation required
- Obviously supports all CSS features out of the box
- Does not support nesting which can be both a blessing (lightweight output) and a curse (impractical sometimes)
- Best tooling support

### PR DESCRIPTION
**Pull request by silverwind**
Ran most of the Less files through the Less compiler and Prettier and then followed up with a round of manual fixes.

The Less compiler had unfortunately stripped all `//` style comments that I had to restore (It did preserve `/* */` comments). Other fixes include duplicate selector removal which were revealed after the transpilation and which weren't caught by stylelint before but now are. 

Fixes: https://github.com/go-gitea/gitea/issues/15565

### CODE REVIEW

**Comment by lunny** [comment]
Thanks, it's a big step toward supporting dynamic themes.

**Comment by wxiaoguang** [comment]
If this PR would be merged sooner or later, I suggest to merge it ASAP to avoid other PRs conflict with this one.

And it's better to kindly remind other PR (which conflicts) authors to pay attention to this change, and kindly thanks them for resolving conflicts manually, or maintainers do the help to resolve the conflicts.

**Comment by silverwind** [comment]
The `.5rem` to `0.5rem` changes are unintended by the way, it's just a result of the prettier formatting. I will check tomorrow if that can be undone by using a now-deprecated stylelint autofixer, but it's not blocking and if you want to land, go ahead.

**Comment by wxiaoguang** [comment]
I like `0.5` more than `.5`. I didn't see why the leading zero must be dropped, it just causes reading difficulty.

**Comment by wxiaoguang** [comment]
I guess many styles are different between 1.19 and 1.20, so a cherry-pick backport won't work. 

If we want to have this in 1.19, just do the same thing on 1.19 branch separately.

**Comment by silverwind** [comment]
> I like 0.5 more than .5. I didn't see why the leading zero must be dropped, it just causes reading difficulty.

Sure, we can keep it. Both formats are fine with current lint config since stylelint had deprecated all their stylistic rules.

File: web_src/js/components/RepoActionView.vue
Comment (delvh) [inline web_src/js/components/RepoActionView.vue]: Should we readd these comments?

File: web_src/js/components/RepoActionView.vue
Comment (delvh) [inline web_src/js/components/RepoActionView.vue]: Ah, we simply threw out this TODO?

**Comment by lunny** [comment]
> And I am almost tempted to say that we should also backport this PR. Otherwise, we basically won't be able to backport a lot of things automatically to 1.19 anymore, basically anything that touches the classes.

Maybe not a good idea. I believe after this merged, some places may need to adjust. For 1.19, it's almost stable and should not have big frontend changes. Frontend bug could have a manual backport?

File: web_src/js/components/RepoActionView.vue
Comment (wxiaoguang) [inline web_src/js/components/RepoActionView.vue]: `height: calc(-166vh);` ?

File: web_src/css/base.css
Comment (delvh) [inline web_src/css/base.css:669]: What happened to this specific line?
That's no longer present…

**Comment by silverwind** [comment]
Yes, this is very likely to introduce a bug or two, so probably should not backport right away at least.

File: web_src/js/components/RepoActionView.vue
Comment (silverwind) [reply web_src/js/components/RepoActionView.vue]: Interesting bug. I ran the Less in Vue through https://jsonformatter.org/less-to-css and it does reproduce the bug.

Comment (wxiaoguang) [reply web_src/js/components/RepoActionView.vue]: Hopefully there isn't other `calc` bug anywhere else.

Comment (silverwind) [reply web_src/js/components/RepoActionView.vue]: Will re-run through `lessc` which does not show that bug

```bash
$ echo 'a{height: calc(100vh - 266px)}' > x.less
$ npx lessc x.less
a {
  height: calc(100vh - 266px);
}
```

Comment (silverwind) [reply web_src/js/components/RepoActionView.vue]: All `.less` files were ran though `lessc`, so only the styles in Vue components can be affected.

**Review by delvh — APPROVED** [review]
Yep, I'm interested in how many bugs this will introduce as seen above.
However, if this PR should ever get merged, it must basically be merged ASAP, so I'm approving for now, even though there are open comments.
I've skimmed over it (I reviewed especially the files were GitHub showed that it was renamed in detail), I probably missed a lot for the completely new files

File: web_src/css/base.css
Comment (silverwind) [reply web_src/css/base.css:669]: `-webkit-user-select` was removed because standard `user-select` is [well supported](https://developer.mozilla.org/en-US/docs/Web/CSS/user-select#browser_compatibility) in all browsers.

The selector itself was moved to line 321 and combined with others that had previoulsy used it via `extend` function  but we can move it back to this place again.

Comment (delvh) [reply web_src/css/base.css:669]: I only meant the `-webkit-user-select`, the position is fairly unimportant for me (as long as we don't agree on some sort of order for selectors)

File: web_src/js/components/RepoActionView.vue
Comment (silverwind) [reply web_src/js/components/RepoActionView.vue]: Yes, the online less compiler used for `.vue` files threw out comments but I will re-run these through `lessc` and check that all comments and empty rules are preserved.

File: web_src/js/components/RepoActionView.vue
Comment (silverwind) [reply web_src/js/components/RepoActionView.vue]: They should be there, excluding the `================` line.

File: web_src/css/base.css
Comment (silverwind) [reply web_src/css/base.css:669]: I guess I can add it back so it's a clean refactor :)

Comment (silverwind) [reply web_src/css/base.css:669]: Restored it.

File: web_src/js/components/RepoActionView.vue
Comment (silverwind) [reply web_src/js/components/RepoActionView.vue]: Rebuilt it, it's fixed and comments are restored as well.

File: web_src/js/components/RepoActionView.vue
Comment (silverwind) [reply web_src/js/components/RepoActionView.vue]: Restored them with the line.

File: web_src/js/components/RepoActionView.vue
Comment (silverwind) [reply web_src/js/components/RepoActionView.vue]: TODOs is back in original style with empty selectors.

**Comment by silverwind** [comment]
Fixed all remaining issues as well as comments that were misplaced by either lessc or prettier.

I agree, we should land rather sooner than later because of merge conflict risk.

**Review by lunny — APPROVED** [review]

File: web_src/css/themes/theme-auto.css
Comment (a1012112796) [inline web_src/css/themes/theme-auto.css:1]: @silverwind looks this style was always be imported...

test with `chorme 111.0.5563.65` on windows 11

屏幕截图(52)

maybe still need use this way 
```CSS
@media (prefers-color-scheme: dark) {
@import "./theme-arc-green.css"
}
```

Comment (silverwind) [reply web_src/css/themes/theme-auto.css:1]: The old variant should not be valid CSS because `@import` can only appear on top-level, that's why I converted it to this new syntax. Running the above through `lessc` yields no difference, so I wonder how this had even worked before.

New syntax follows recommendation from [this](https://blog.jim-nielsen.com/2019/conditional-syntax-highlighting-in-dark-mode-with-css-imports/), so I'm surprised it does not work.

Comment (silverwind) [reply web_src/css/themes/theme-auto.css:1]: Ah, I see the issue, the new `theme-auto` compilation stripped the media query off, resulting in arc-green always being loaded.

So my point stands true that above is invalid CSS in itself, but with webpack's `import` compilation actually makes it valid by inlining the CSS into the media query.

Comment (silverwind) [reply web_src/css/themes/theme-auto.css:1]: https://github.com/go-gitea/gitea/pull/23504 will fix it

**Pull request merged**
